### PR TITLE
New plugin added: Leaflet.TextPath

### PR DESCRIPTION
**Pull request by talespaiva**
Port of Leaflet.TextPath plugin: [https://github.com/makinacorpus/Leaflet.TextPath](https://github.com/makinacorpus/Leaflet.TextPath)

Example

Also added examples in the plugins Jupyter notebook.

### CODE REVIEW

File: examples/polyline_text_path_example.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+# This is to import the repository's version of folium ; not the installed one.
+import sys, os
+sys.path.insert(0, '..')
```
Comment (ocefpaf) [inline examples/polyline_text_path_example.py]: I think we can drop these two lines. (BTW I am planning to re-do all examples as notebooks that we can run on Travis as "visual"  tests.)

File: folium/plugins/polyline_text_path.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                Set an offset to position text relative to the polyline.
+
+            orientation: int, default 0
+                Rotate text to a specified angle.
+
+            attributes: dictionary
+                Object containing the attributes applied to the text tag.
+                Check valid attributes here: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/text#Attributes
+                Example: {'fill': '#007DEF', 'font-weight': 'bold', 'font-size': '24'}
+
```
Comment (ocefpaf) [inline folium/plugins/polyline_text_path.py]: Please add a note and the URL for the plugin page. (I guess https://github.com/makinacorpus/Leaflet.TextPath ?)

**Comment by ocefpaf** [comment]
> Port of Leaflet.TextPath plugin: https://github.com/makinacorpus/Leaflet.TextPath

Cool!

> Also added examples in the plugins Jupyter notebook.

Awesome!!

I made a minor comment and this LGTM (the failures are only styling/pep8ing).

@BibMartin I will let this sit until next week so you get a change to look at it.

Thanks @talespaiva!!!

**Comment by ocefpaf** [comment]
@talespaiva can you please add an entry in the [changelog](https://github.com/python-visualization/folium/blob/master/CHANGES.txt#L5).

**Comment by BibMartin** [comment]
And fix the pep8 stuff. According to travis:

```
./examples/polyline_text_path_example.py:5:20: E231 missing whitespace after ','
./examples/polyline_text_path_example.py:6:101: E501 line too long (107 > 100 characters)
./examples/polyline_text_path_example.py:7:14: E128 continuation line under-indented for visual indent
./examples/polyline_text_path_example.py:8:101: E501 line too long (106 > 100 characters)
./examples/polyline_text_path_example.py:9:101: E501 line too long (106 > 100 characters)
./examples/polyline_text_path_example.py:15:101: E501 line too long (121 > 100 characters)
./examples/polyline_text_path_example.py:19:101: E501 line too long (135 > 100 characters)
./examples/polyline_text_path_example.py:20:22: E231 missing whitespace after ':'
./examples/polyline_text_path_example.py:20:42: E231 missing whitespace after ':'
./examples/polyline_text_path_example.py:21:101: E501 line too long (101 > 100 characters)
./folium/plugins/polyline_text_path.py:19:101: E501 line too long (120 > 100 characters)
./folium/plugins/polyline_text_path.py:47:101: E501 line too long (117 > 100 characters)
./tests/plugins/test_polyline_text_path.py:16:101: E501 line too long (111 > 100 characters)
./tests/plugins/test_polyline_text_path.py:17:101: E501 line too long (106 > 100 characters)
./tests/plugins/test_polyline_text_path.py:18:101: E501 line too long (115 > 100 characters)
./tests/plugins/test_polyline_text_path.py:19:101: E501 line too long (115 > 100 characters)
./tests/plugins/test_polyline_text_path.py:22:5: E303 too many blank lines (2)
./tests/plugins/test_polyline_text_path.py:24:101: E501 line too long (101 > 100 characters)
```

**Comment by ocefpaf** [comment]
@talespaiva this needs a rebase and some minor style fixes. I am planning a new release soon, so let me know if you have any plans on finishing this or if it is alright to pick your commits, finished it, and close this PR.

**Comment by talespaiva** [comment]
@ocefpaf It's alright to pick my commits and finish it. I can't work in this one right now. I hope it's ok.

**Pull request closed without merging**